mlir/include/mlir/IR/BuiltinAttributes.h

```diff
@@ -91,7 +91,7 @@ class DenseElementsAttr : public Attribute {
     /// point type, or is a potentially user defined floating point type. The
     /// latter allows for supporting users that have custom types defined for
     /// bfloat16/half/etc.
-    static constexpr bool value = llvm::is_one_of<T, float, double>::value ||
+    static constexpr bool value = llvm::is_one_of<T, float, double, __bf16>::value ||
```

Comment:
This would be the first use of this compiler-specific data type. Do you expect this to be supported upstream of is this something we must maintain out of tree?